### PR TITLE
[AKS] Add `--enable-azure-monitor-app-monitoring` and `--disable-azure-monitor-app-monitoring` to aks-preview

### PR DESCRIPTION
**Pull request by jatakiajanvi12**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
aks update --resource-group={resource_group} --name={name} ---enable-azure-monitor-app-monitoring
aks update --resource-group={resource_group} --name={name} ---disable-azure-monitor-app-monitoring

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/aks-preview/linter_exclusions.yml [hunk 1]
```diff
@@ -75,6 +75,9 @@ aks create:
     bootstrap_container_registry_resource_id:
       rule_exclusions:
         - option_length_too_long
+    enable_azure_monitor_app_monitoring:
+      rule_extensions:
```
Comment (FumingZhang) [inline src/aks-preview/linter_exclusions.yml:79]:
```suggestion
      rule_exclusions:
```

File: src/aks-preview/linter_exclusions.yml [hunk 2]
```diff
@@ -173,6 +176,12 @@ aks update:
     disable_ai_toolchain_operator:
       rule_exclusions:
         - option_length_too_long
+    enable_azure_monitor_app_monitoring:
+      rule_extensions:
```
Comment (FumingZhang) [inline src/aks-preview/linter_exclusions.yml:180]:
```suggestion
      rule_exclusions:
```

File: src/aks-preview/linter_exclusions.yml [hunk 3]
```diff
@@ -173,6 +176,12 @@ aks update:
...
+        - option_length_too_long
+    disable_azure_monitor_app_monitoring:
+      rule_extensions:
```
Comment (FumingZhang) [inline src/aks-preview/linter_exclusions.yml:183]:
```suggestion
      rule_exclusions: 
```

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 4]
```diff
@@ -10610,6 +10610,59 @@ def test_aks_create_with_azuremonitormetrics(
             ],
         )
 
+    # live only due to downloading k8s-extension extension
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:10613]: > # live only due to downloading k8s-extension extension

This comment is misleading.

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 5]
```diff
@@ -10610,6 +10610,59 @@ def test_aks_create_with_azuremonitormetrics(
...
+    @live_only()
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:10614]: consider remove `@live_only()` if possible

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 6]
```diff
@@ -10787,6 +10840,59 @@ def test_aks_update_with_azuremonitormetrics(
             ],
         )
 
+    @live_only()
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:10843]: consider remove `@live_only()` if possible

File: src/aks-preview/azext_aks_preview/_params.py [hunk 7]
```diff
@@ -1206,6 +1207,8 @@ def load_arguments(self, _):
             ),
         )
         c.argument("disable_azure_monitor_metrics", action="store_true")
+        c.argument("enable_azure_monitor_app_monitoring", action="store_true")
+        c.argument("disable_azure_monitor_app_monitoring", action="store_true")
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:1211]: set `is_preview=True` for these 2 options too?

**Review by FumingZhang — COMMENTED** [review]
Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=91688820&view=results) to validate the change
- test_aks_create_with_azuremonitorappmonitoring
- test_aks_update_with_azuremonitorappmonitoring

**Review by tokaplan — APPROVED** [review]

**Review by FumingZhang — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/_help.py [hunk 8]
```diff
@@ -698,6 +701,8 @@
           text: az aks create -g MyResourceGroup -n MyManagedCluster --enable-azure-service-mesh
         - name: Create a kubernetes cluster with Azure Monitor Metrics enabled.
           text: az aks create -g MyResourceGroup -n MyManagedCluster --enable-azuremonitormetrics
+        - name: Create a kubernetes cluster with Azure Monitor App Monitoring enabled
+          text: az aks create -g MyResourceGroup -n MyManagedCluster --enable-azure-monitor-app-monitoring
```
Comment (ganga1980) [inline src/aks-preview/azext_aks_preview/_help.py:705]: @jatakiajanvi12 , are we not exposing the open telemetry flags and ports to configure?

Comment (ganga1980) [inline src/aks-preview/azext_aks_preview/_help.py:705]: I dont see preview API version. Did we test these changes for both create and update scenarios?

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 9]
```diff
@@ -3104,6 +3155,23 @@ def set_up_azure_monitor_profile(self, mc: ManagedCluster) -> ManagedCluster:
...
                 metric_annotations_allow_list=str(ksm_metric_annotations_allow_list))
             self.context.set_intermediate("azuremonitormetrics_addon_enabled", True, overwrite_exists=True)
+
+        if self.context.get_enable_azure_monitor_app_monitoring():
+            if mc.azure_monitor_profile is None:
+                mc.azure_monitor_profile = self.models.ManagedClusterAzureMonitorProfile()
+            mc.azure_monitor_profile.app_monitoring = (
+                self.models.ManagedClusterAzureMonitorProfileAppMonitoring()
+            )
+            mc.azure_monitor_profile.app_monitoring.auto_instrumentation = (
```
Comment (ganga1980) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:3165]: can customer specify these options as part of the onboarding?

File: src/aks-preview/azext_aks_preview/_help.py (hunk 8 above)
Comment (jatakiajanvi12) [reply src/aks-preview/azext_aks_preview/_help.py:705]: Will be doing that in a separate PR

Comment (jatakiajanvi12) [reply src/aks-preview/azext_aks_preview/_help.py:705]: Yes both the scenarios have been tested. There are live tests for the same. The default api version of the aks preview is the latest preview api

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 9 above)
Comment (jatakiajanvi12) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:3165]: Yes they can specify this

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**